src/Adyen/Client.php

```diff
@@ -89,24 +91,32 @@ public function setXApiKey($xApiKey)
 
     /**
      * Set environment to connect to test or live platform of Adyen
+     * For live please specify the unique identifier.
      *
      * @param $environment
+     * @param null $liveEndpointUrlPrefix
      * @throws AdyenException
      */
-    public function setEnvironment($environment)
+    public function setEnvironment($environment, $liveEndpointUrlPrefix = null)
     {
         if ($environment == \Adyen\Environment::TEST) {
             $this->_config->set('environment', \Adyen\Environment::TEST);
             $this->_config->set('endpoint', self::ENDPOINT_TEST);
             $this->_config->set('endpointDirectorylookup', self::ENDPOINT_TEST_DIRECTORY_LOOKUP);
-            $this->_config->set('endpointTerminalCloud', self::TERMINAL_CLOUD_TEST);
+            $this->_config->set('endpointTerminalCloud', self::ENDPOINT_TERMINAL_CLOUD_TEST);
             $this->_config->set('endpointCheckout', self::ENDPOINT_CHECKOUT_TEST);
         } elseif ($environment == \Adyen\Environment::LIVE) {
             $this->_config->set('environment', \Adyen\Environment::LIVE);
-            $this->_config->set('endpoint', self::ENDPOINT_LIVE);
             $this->_config->set('endpointDirectorylookup', self::ENDPOINT_LIVE_DIRECTORY_LOOKUP);
-            $this->_config->set('endpointTerminalCloud', self::TERMINAL_CLOUD_LIVE);
-            $this->_config->set('endpointCheckout', self::ENDPOINT_CHECKOUT_LIVE);
+            $this->_config->set('endpointTerminalCloud', self::ENDPOINT_TERMINAL_CLOUD_LIVE);
+
+            if ($liveEndpointUrlPrefix) {
+                $this->_config->set('endpoint', self::ENDPOINT_PROTOCOL . $liveEndpointUrlPrefix . self::ENDPOINT_LIVE_SUFFIX);
+                $this->_config->set('endpointCheckout', self::ENDPOINT_PROTOCOL . $liveEndpointUrlPrefix . self::ENDPOINT_CHECKOUT_LIVE_SUFFIX);
+            } else {
+                $this->_config->set('endpoint', self::ENDPOINT_LIVE);
+                $this->_config->set('endpointCheckout', null); // not supported please specify unique identifier
```

Comment:
minor: let's not set it to null, in case they specify this value "endpointCheckout " before setEnvironment call

Reply:
if you first provide setEnvironment TEST and override with LIVE this result in a TEST endpoint that is why i specifically set it to null. Let's verify your use case tomorrow.